File: 2578. Split With Minimum Sum/Solution.py
class Solution:
    def splitNum(self, num: int) -> int:
        digits = []
        while num > 0:
            digits.append(num % 10)
            num //= 10
        digits = sorted(digits)
        n = len(digits)
        res = digits[0] if n % 2 == 1 else 0
        pointer = 1 if n % 2 == 1 else 0
        while pointer < n:
            res = (res * 10) + digits[pointer] + digits[pointer + 1]
            pointer += 2
        return res

File: 2578. Split With Minimum Sum/Solution.java
class Solution {
    public int splitNum(int num) {
        List<Integer> digits = new ArrayList<>();
        while (num > 0) {
            digits.add(num % 10);
            num /= 10;
        }
        Collections.sort(digits);
        int res = 0, pointer = 0;
        if (digits.size() % 2 == 1) {
            res += digits.get(pointer);
            pointer++;
        }
        while (pointer < digits.size()) {
            res = res * 10 + digits.get(pointer) + digits.get(pointer + 1);
            pointer += 2;
        }
        return res;
    }
}